src/main/java/com/juniorstart/juniorstart/service/UserService.java

```diff
@@ -34,4 +37,24 @@ public User getCurrentUser(@CurrentUser UserPrincipal userPrincipal) {
                 .orElseThrow(() -> new ResourceNotFoundException("User", "id", userPrincipal.getId()));
     }
 
+    /** Update user email.
+     * @param email new email.
+     * @param username login for user
+     * @param password password for user
+     * @return a current user credentials
+     * @throws RuntimeException if email is to long or cannot find user.
+     */
+    public boolean changeEmail(@Email String email, String username, String password) {
+        if (email.length() <= 40) {
```

Comment:
Spoko pomysł. żeby sprawdzić, ale można oszczędzić tej linijki przekazując obiekt, wtedy automatycznie wyskoczy błąd przy braku jakiegoś pola. Przykład podobnej metody. Możesz się zainspirować. 


 public ResponseEntity<?> updateName(UserPrincipal currentUser, UpdateNameRequest updateNameRequest) {
       User user = userRepository.findById(currentUser.getId())
               .orElseThrow(() -> new ResourceNotFoundException("User", "id", currentUser.getId()));

       if(userRepository.existsByName(updateNameRequest.getName())) {
           return new ResponseEntity(new ApiResponse(false, "Name already in user"), HttpStatus.BAD_REQUEST);
       }

       user.setName(updateNameRequest.getName());

       User result =  userRepository.save(user);

       return ResponseEntity.ok().body(new ApiResponse(true, "User name updated successfully"));

    }